src/cpp/middleware/fast/FastMiddleware.cpp

```diff
@@ -88,33 +93,32 @@ bool FastMiddleware::create_topic_from_xml(uint16_t topic_id, uint16_t participa
         fastrtps::TopicAttributes attributes;
         if (xmlobjects::parse_topic(xml.data(), xml.size(), attributes))
         {
-            std::shared_ptr<FastTopic> topic(new FastTopic());
-            if (topic->create_by_attributes(attributes, it_participant->second.get()))
+            std::shared_ptr<FastTopic> topic(new FastTopic(it_participant->second));
+            if (topic->create_by_attributes(attributes, topic_id))
             {
-                register_topic(topic->getName(), topic_id);
```

Comment:
Why do you not register it anymore?

Reply:
That is something I have to change in the near future. Added TODO with a task.